plugins/woocommerce/src/Blocks/BlockTypes/MiniCartFooterBlock.php

```diff
@@ -92,6 +102,67 @@ protected function get_cart_instance() {
 		return null;
 	}
 
+	/**
+	 * Computes the total items description text based on which settings are enabled.
+	 *
+	 * @return string The description text for the total items, or empty string if none are enabled.
+	 */
+	private function get_total_items_description() {
+		$taxes_enabled    = wc_tax_enabled();
+		$shipping_enabled = wc_shipping_enabled();
+		$coupons_enabled  = wc_coupons_enabled();
+
+		// All three enabled.
+		if ( $taxes_enabled && $shipping_enabled && $coupons_enabled ) {
+			return __(
+				'Shipping, taxes, and discounts calculated at checkout.',
+				'woocommerce'
+			);
+		}
+
+		// Shipping + taxes.
+		if ( $shipping_enabled && $taxes_enabled ) {
+			return __(
+				'Shipping and taxes calculated at checkout.',
+				'woocommerce'
+			);
+		}
+
+		// Shipping + discounts.
+		if ( $shipping_enabled && $coupons_enabled ) {
+			return __(
+				'Shipping and discounts calculated at checkout.',
+				'woocommerce'
+			);
+		}
+
+		// Taxes + discounts.
+		if ( $taxes_enabled && $coupons_enabled ) {
+			return __(
+				'Taxes and discounts calculated at checkout.',
+				'woocommerce'
+			);
+		}
+
+		// Only shipping.
+		if ( $shipping_enabled ) {
+			return __( 'Shipping calculated at checkout.', 'woocommerce' );
+		}
+
+		// Only taxes.
+		if ( $taxes_enabled ) {
+			return __( 'Taxes calculated at checkout.', 'woocommerce' );
+		}
+
+		// Only discounts.
+		if ( $coupons_enabled ) {
+			return __( 'Discounts calculated at checkout.', 'woocommerce' );
+		}
```

Comment:
A bummer we need all those new strings, but at the same time, the extra complexity if we were to reduce the repetition here and have fewer strings is likely not worth it.

Reply:
Ya, and for i18n purposes there isn't much we can do dynamically without risking improper context communicated to translators.